static/app/components/replays/videoReplayer.tsx

```diff
@@ -194,31 +308,51 @@ export class VideoReplayer {
     return this._videos[index];
   }
 
-  protected hideVideo(index: number | undefined): void {
-    const video = this.getVideo(index);
-
-    if (!video) {
+  /**
+   * Shows the video -- it is assumed that it is preloaded. Also
+   * hides the previous video, there should not be a reason we show
+   * a video and not hide the previous video, otherwise there will
+   * be multiple video elements stacked on top of each other.
+   */
+  protected showVideo(nextVideo: HTMLVideoElement | undefined): void {
+    if (!nextVideo) {
       return;
     }
 
-    video.style.display = 'none';
-  }
-
-  protected showVideo(video: HTMLVideoElement | undefined): void {
-    if (!video) {
-      return;
+    // This is the soon-to-be previous video that needs to be hidden
+    if (this._currentVideo) {
+      this._currentVideo.style.display = 'none';
     }
 
     // TODO: resize video if it changes orientation
-    video.style.display = 'block';
+    nextVideo.style.display = 'block';
+
+    // Update current video so that we can hide it when showing the
+    // next video
+    this._currentVideo = nextVideo;
   }
 
-  protected playVideo(video: HTMLVideoElement | undefined): Promise<void> {
+  protected async playVideo(video: HTMLVideoElement | undefined): Promise<void> {
     if (!video) {
       return Promise.resolve();
     }
     video.playbackRate = this.config.speed;
-    return video.play();
+
+    // If video is not playable, then update buffering state,
+    // otherwise, proceed to hide the previous video and play
+    if (video.readyState === 0) {
+      // Note that we do not handle when the load finishes here, it
+      // is handled via the `loadeddata` event handler
+      this.setBuffering(true);
+    }
+
+    const playPromise = video.play();
+    await playPromise;
+
+    // Buffering is over after play promise is resolved
+    this.setBuffering(false);
+
+    return playPromise;
```

Comment:
Don't need to return that specific `void`, the whole function will still return type `Promise<void>`

```suggestion
    await video.play();

    // Buffering is over after play promise is resolved
    this.setBuffering(false);
```

Reply:
maybe the return keyword is needed then... something about it look funny still though but low gaf level, it's not broken.